zipkin-storage/elasticsearch/src/test/java/zipkin2/elasticsearch/VersionSpecificTemplatesTest.java

```diff
@@ -102,56 +98,44 @@
       + "  \"tagline\" : \"You Know, for Search\"\n"
       + "}");
 
-  static final AtomicReference<AggregatedHttpResponse> MOCK_RESPONSE =
-    new AtomicReference<>();
+  @RegisterExtension static MockWebServerExtension server = new MockWebServerExtension();
 
-  @Rule public ExpectedException thrown = ExpectedException.none();
-
-  @ClassRule public static ServerRule server = new ServerRule() {
-    @Override protected void configure(ServerBuilder sb) {
-      sb.serviceUnder("/", (ctx, req) -> HttpResponse.of(MOCK_RESPONSE.get()));
-    }
-  };
-
-  @Before public void setUp() {
+  @BeforeEach void setUp() {
     storage = ElasticsearchStorage.newBuilder(() -> HttpClient.of(server.httpUri("/"))).build();
   }
 
-  @After public void tearDown() {
+  @AfterEach void tearDown() {
     storage.close();
   }
 
   ElasticsearchStorage storage;
 
-  @Test public void wrongContent() throws Exception {
-    MOCK_RESPONSE.set(AggregatedHttpResponse.of(
+  @Test void wrongContent() throws Exception {
+    server.enqueue(AggregatedHttpResponse.of(
       ResponseHeaders.of(HttpStatus.OK),
       HttpData.ofUtf8("you got mail")));
 
-    thrown.expectMessage(".version.number not found in response: you got mail");
-
-    new VersionSpecificTemplates(storage).get();
+    assertThatThrownBy(() -> new VersionSpecificTemplates(storage).get())
```

Comment:
PS eyes are finally used to this :) besides one wonky thing I've not hunted down (eats StackOverFlowError), in general I like this better than the thrown rule and manually try/catch